src/hotspot/share/logging/logAsyncWriter.cpp

```diff
@@ -81,73 +118,93 @@ AsyncLogWriter::AsyncLogWriter()
   : _flush_sem(0), _lock(), _data_available(false),
     _initialized(false),
     _stats() {
+
+  size_t page_size = os::vm_page_size();
+  size_t size = align_up(AsyncLogBufferSize / 2, page_size);
+
+  char* buf0 = static_cast<char* >(os::malloc(size, mtLogging));
+  char* buf1 = static_cast<char* >(os::malloc(size, mtLogging));
+  if (buf0 != nullptr && buf1 != nullptr) {
+    _buffer = new Buffer(buf0, size);
+    _buffer_staging = new Buffer(buf1, size);
+    if (_buffer != nullptr && _buffer_staging != nullptr) {
+      log_info(logging)("AsyncLogBuffer estimates memory use: " SIZE_FORMAT " bytes", size * 2);
+    } else {
+      log_warning(logging)("AsyncLogging failed to create buffer Objects. Falling back to synchronous logging.");
+      delete _buffer;
+      delete _buffer_staging;
+      os::free(buf0);
+      os::free(buf1);
+      return;
+    }
+  } else {
+    log_warning(logging)("AsyncLogging failed to create buffers. Falling back to synchronous logging.");
+
+    if (buf0 != nullptr) {
```

Comment:
`os::free(nullptr)` is safe, no need for a branch here.

Reply:
okay! I will update it.